core/lib/spree/core/controller_helpers/order.rb

```diff
@@ -37,6 +37,11 @@ def current_order(options = {})
         end
 
         def associate_user
+          return if Spree::Config.associate_user_in_authentication_extension
```

Comment:
I think we should just stop calling this method from inside our core codebase instead of returning if the preference is true.

I can see at least a couple of occurrences:

- https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/frontend/app/controllers/spree/orders_controller.rb#L43
- https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/frontend/app/controllers/spree/checkout_controller.rb#L18


If it's useless, things should keep working as before.

Reply:
I think adding a conditional on the value of the preference in multiple places may be slightly worse than just checking it in the method itself.

Reply:
@forkata I believe @kennyadsl is suggesting not having an option at all. We can simply deprecate it and stop using it in core, then later remove it entirely.

Reply:
That makes, sense, I misread the comment here 😅

Reply:
@waiting-for-dev @forkata @kennyadsl Please read my comment above https://github.com/solidusio/solidus/pull/4201#issuecomment-982833893